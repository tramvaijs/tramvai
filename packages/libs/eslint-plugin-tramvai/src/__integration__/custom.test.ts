const path = require('node:path');

describe('custom eslint plugin tramvai', () => {
  it('should return default config with tramvai rules', () => {
    const compatConfig = require('./__fixtures__/default/eslint-plugin-tramvai-compat');

    expect(compatConfig).toMatchSnapshot();
  });

  it('should return custom config with polyfills from default', () => {
    const compatConfig = require('./__fixtures__/custom-default/eslint-plugin-tramvai-compat');

    expect(compatConfig.rules['es-x/no-promise-withresolvers']).toBe(0);
    expect(compatConfig.rules['es-x/no-float16array']).toBe(0);
  });

  it('should return custom config with polyfills from modern', () => {
    const compatConfig = require('./__fixtures__/modern/eslint-plugin-tramvai-compat');

    expect(compatConfig.rules['es-x/no-string-prototype-iswellformed-towellformed']).toBe(0);
  });

  it('should return custom config with custom browserslist', () => {
    const compatConfig = require('./__fixtures__/custom-browserslist/eslint-plugin-tramvai-compat');

    expect(compatConfig.rules['es-x/no-weakrefs']).toBe(0);
  });

  it('should return custom config with custom package', () => {
    const compatConfig = require('./__fixtures__/custom-package/eslint-plugin-tramvai-compat');

    // check web polyfil import
    expect(compatConfig.settings.polyfills.includes('structuredClone')).toBeTruthy();

    // check custom rule usage
    expect(compatConfig.rules['es-x/no-atomics']).toBe(0);

    // check modern browserslist config usage
    expect(compatConfig.rules['es-x/no-weakrefs']).toBe(0);
  });
});
