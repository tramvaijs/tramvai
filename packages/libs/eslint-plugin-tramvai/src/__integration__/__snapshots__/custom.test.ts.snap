// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`custom eslint plugin tramvai should return default config with tramvai rules 1`] = `
{
  "plugins": [
    "es-x",
    "compat",
  ],
  "rules": {
    "compat/compat": "error",
    "es-x/no-accessor-properties": 0,
    "es-x/no-arbitrary-module-namespace-names": 0,
    "es-x/no-array-from": 0,
    "es-x/no-array-fromasync": 0,
    "es-x/no-array-isarray": 0,
    "es-x/no-array-of": 0,
    "es-x/no-array-prototype-at": 0,
    "es-x/no-array-prototype-copywithin": 0,
    "es-x/no-array-prototype-entries": 0,
    "es-x/no-array-prototype-every": 0,
    "es-x/no-array-prototype-fill": 0,
    "es-x/no-array-prototype-filter": 0,
    "es-x/no-array-prototype-find": 0,
    "es-x/no-array-prototype-findindex": 0,
    "es-x/no-array-prototype-findlast-findlastindex": 2,
    "es-x/no-array-prototype-flat": 0,
    "es-x/no-array-prototype-foreach": 0,
    "es-x/no-array-prototype-includes": 0,
    "es-x/no-array-prototype-indexof": 0,
    "es-x/no-array-prototype-keys": 0,
    "es-x/no-array-prototype-lastindexof": 0,
    "es-x/no-array-prototype-map": 0,
    "es-x/no-array-prototype-reduce": 0,
    "es-x/no-array-prototype-reduceright": 0,
    "es-x/no-array-prototype-some": 0,
    "es-x/no-array-prototype-toreversed": 2,
    "es-x/no-array-prototype-tosorted": 2,
    "es-x/no-array-prototype-tospliced": 2,
    "es-x/no-array-prototype-values": 0,
    "es-x/no-array-prototype-with": 2,
    "es-x/no-array-string-prototype-at": 0,
    "es-x/no-arraybuffer-prototype-transfer": 2,
    "es-x/no-arrow-functions": 0,
    "es-x/no-async-functions": 0,
    "es-x/no-async-iteration": 0,
    "es-x/no-asyncdisposablestack": 0,
    "es-x/no-atomics": 2,
    "es-x/no-atomics-waitasync": 2,
    "es-x/no-bigint": 0,
    "es-x/no-binary-numeric-literals": 0,
    "es-x/no-block-scoped-functions": 0,
    "es-x/no-block-scoped-variables": 0,
    "es-x/no-class-fields": 0,
    "es-x/no-class-instance-fields": 0,
    "es-x/no-class-private-fields": 0,
    "es-x/no-class-private-methods": 0,
    "es-x/no-class-static-block": 0,
    "es-x/no-class-static-fields": 0,
    "es-x/no-classes": 0,
    "es-x/no-computed-properties": 0,
    "es-x/no-dataview-prototype-getfloat16-setfloat16": 2,
    "es-x/no-date-now": 0,
    "es-x/no-date-prototype-getyear-setyear": 0,
    "es-x/no-date-prototype-togmtstring": 0,
    "es-x/no-default-parameters": 0,
    "es-x/no-destructuring": 0,
    "es-x/no-disposablestack": 0,
    "es-x/no-dynamic-import": 0,
    "es-x/no-dynamic-import-options": 0,
    "es-x/no-error-cause": 2,
    "es-x/no-error-iserror": 0,
    "es-x/no-escape-unescape": 0,
    "es-x/no-exponential-operators": 0,
    "es-x/no-export-ns-from": 0,
    "es-x/no-float16array": 2,
    "es-x/no-for-of-loops": 0,
    "es-x/no-function-declarations-in-if-statement-clauses-without-block": 0,
    "es-x/no-function-prototype-bind": 0,
    "es-x/no-generators": 0,
    "es-x/no-global-this": 0,
    "es-x/no-hashbang": 0,
    "es-x/no-import-attributes": 0,
    "es-x/no-import-meta": 0,
    "es-x/no-initializers-in-for-in": 0,
    "es-x/no-intl-datetimeformat-prototype-formatrange": 2,
    "es-x/no-intl-datetimeformat-prototype-formattoparts": 2,
    "es-x/no-intl-displaynames": 2,
    "es-x/no-intl-durationformat": 2,
    "es-x/no-intl-getcanonicallocales": 0,
    "es-x/no-intl-listformat": 2,
    "es-x/no-intl-locale": 0,
    "es-x/no-intl-numberformat-prototype-formatrange": 2,
    "es-x/no-intl-numberformat-prototype-formatrangetoparts": 2,
    "es-x/no-intl-numberformat-prototype-formattoparts": 0,
    "es-x/no-intl-pluralrules": 0,
    "es-x/no-intl-pluralrules-prototype-selectrange": 2,
    "es-x/no-intl-relativetimeformat": 0,
    "es-x/no-intl-segmenter": 2,
    "es-x/no-intl-supportedvaluesof": 2,
    "es-x/no-iterator": 0,
    "es-x/no-iterator-prototype-drop": 2,
    "es-x/no-iterator-prototype-every": 2,
    "es-x/no-iterator-prototype-filter": 2,
    "es-x/no-iterator-prototype-find": 2,
    "es-x/no-iterator-prototype-flatmap": 2,
    "es-x/no-iterator-prototype-foreach": 2,
    "es-x/no-iterator-prototype-map": 2,
    "es-x/no-iterator-prototype-reduce": 2,
    "es-x/no-iterator-prototype-some": 2,
    "es-x/no-iterator-prototype-take": 2,
    "es-x/no-iterator-prototype-toarray": 2,
    "es-x/no-json": 0,
    "es-x/no-json-modules": 0,
    "es-x/no-json-superset": 0,
    "es-x/no-keyword-properties": 0,
    "es-x/no-labelled-function-declarations": 0,
    "es-x/no-legacy-object-prototype-accessor-methods": 0,
    "es-x/no-logical-assignment-operators": 0,
    "es-x/no-malformed-template-literals": 0,
    "es-x/no-map": 0,
    "es-x/no-map-groupby": 2,
    "es-x/no-math-acosh": 0,
    "es-x/no-math-asinh": 0,
    "es-x/no-math-atanh": 0,
    "es-x/no-math-cbrt": 0,
    "es-x/no-math-clz32": 0,
    "es-x/no-math-cosh": 0,
    "es-x/no-math-expm1": 0,
    "es-x/no-math-f16round": 2,
    "es-x/no-math-fround": 0,
    "es-x/no-math-hypot": 0,
    "es-x/no-math-imul": 0,
    "es-x/no-math-log10": 0,
    "es-x/no-math-log1p": 0,
    "es-x/no-math-log2": 0,
    "es-x/no-math-sign": 0,
    "es-x/no-math-sinh": 0,
    "es-x/no-math-tanh": 0,
    "es-x/no-math-trunc": 0,
    "es-x/no-modules": 0,
    "es-x/no-new-target": 0,
    "es-x/no-nonstandard-array-properties": 0,
    "es-x/no-nonstandard-array-prototype-properties": 0,
    "es-x/no-nonstandard-arraybuffer-properties": 0,
    "es-x/no-nonstandard-arraybuffer-prototype-properties": 0,
    "es-x/no-nonstandard-asyncdisposablestack-properties": 0,
    "es-x/no-nonstandard-asyncdisposablestack-prototype-properties": 0,
    "es-x/no-nonstandard-atomics-properties": 0,
    "es-x/no-nonstandard-bigint-properties": 0,
    "es-x/no-nonstandard-bigint-prototype-properties": 0,
    "es-x/no-nonstandard-boolean-properties": 0,
    "es-x/no-nonstandard-boolean-prototype-properties": 0,
    "es-x/no-nonstandard-dataview-properties": 0,
    "es-x/no-nonstandard-dataview-prototype-properties": 0,
    "es-x/no-nonstandard-date-properties": 0,
    "es-x/no-nonstandard-date-prototype-properties": 0,
    "es-x/no-nonstandard-disposablestack-properties": 0,
    "es-x/no-nonstandard-disposablestack-prototype-properties": 0,
    "es-x/no-nonstandard-error-properties": 0,
    "es-x/no-nonstandard-finalizationregistry-properties": 0,
    "es-x/no-nonstandard-finalizationregistry-prototype-properties": 0,
    "es-x/no-nonstandard-function-properties": 0,
    "es-x/no-nonstandard-intl-collator-properties": 0,
    "es-x/no-nonstandard-intl-collator-prototype-properties": 0,
    "es-x/no-nonstandard-intl-datetimeformat-properties": 0,
    "es-x/no-nonstandard-intl-datetimeformat-prototype-properties": 0,
    "es-x/no-nonstandard-intl-displaynames-properties": 0,
    "es-x/no-nonstandard-intl-displaynames-prototype-properties": 0,
    "es-x/no-nonstandard-intl-durationformat-properties": 0,
    "es-x/no-nonstandard-intl-durationformat-prototype-properties": 0,
    "es-x/no-nonstandard-intl-listformat-properties": 0,
    "es-x/no-nonstandard-intl-listformat-prototype-properties": 0,
    "es-x/no-nonstandard-intl-locale-properties": 0,
    "es-x/no-nonstandard-intl-locale-prototype-properties": 0,
    "es-x/no-nonstandard-intl-numberformat-properties": 0,
    "es-x/no-nonstandard-intl-numberformat-prototype-properties": 0,
    "es-x/no-nonstandard-intl-pluralrules-properties": 0,
    "es-x/no-nonstandard-intl-pluralrules-prototype-properties": 0,
    "es-x/no-nonstandard-intl-properties": 0,
    "es-x/no-nonstandard-intl-relativetimeformat-properties": 0,
    "es-x/no-nonstandard-intl-relativetimeformat-prototype-properties": 0,
    "es-x/no-nonstandard-intl-segmenter-properties": 0,
    "es-x/no-nonstandard-intl-segmenter-prototype-properties": 0,
    "es-x/no-nonstandard-iterator-properties": 0,
    "es-x/no-nonstandard-iterator-prototype-properties": 0,
    "es-x/no-nonstandard-json-properties": 0,
    "es-x/no-nonstandard-map-properties": 0,
    "es-x/no-nonstandard-map-prototype-properties": 0,
    "es-x/no-nonstandard-math-properties": 0,
    "es-x/no-nonstandard-number-properties": 0,
    "es-x/no-nonstandard-number-prototype-properties": 0,
    "es-x/no-nonstandard-object-properties": 0,
    "es-x/no-nonstandard-promise-properties": 0,
    "es-x/no-nonstandard-promise-prototype-properties": 0,
    "es-x/no-nonstandard-proxy-properties": 0,
    "es-x/no-nonstandard-reflect-properties": 0,
    "es-x/no-nonstandard-regexp-properties": 0,
    "es-x/no-nonstandard-regexp-prototype-properties": 0,
    "es-x/no-nonstandard-set-properties": 0,
    "es-x/no-nonstandard-set-prototype-properties": 0,
    "es-x/no-nonstandard-sharedarraybuffer-properties": 0,
    "es-x/no-nonstandard-sharedarraybuffer-prototype-properties": 0,
    "es-x/no-nonstandard-string-properties": 0,
    "es-x/no-nonstandard-string-prototype-properties": 0,
    "es-x/no-nonstandard-symbol-properties": 0,
    "es-x/no-nonstandard-symbol-prototype-properties": 0,
    "es-x/no-nonstandard-typed-array-properties": 0,
    "es-x/no-nonstandard-typed-array-prototype-properties": 0,
    "es-x/no-nonstandard-weakmap-properties": 0,
    "es-x/no-nonstandard-weakmap-prototype-properties": 0,
    "es-x/no-nonstandard-weakref-properties": 0,
    "es-x/no-nonstandard-weakref-prototype-properties": 0,
    "es-x/no-nonstandard-weakset-properties": 0,
    "es-x/no-nonstandard-weakset-prototype-properties": 0,
    "es-x/no-nullish-coalescing-operators": 0,
    "es-x/no-number-epsilon": 0,
    "es-x/no-number-isfinite": 0,
    "es-x/no-number-isinteger": 0,
    "es-x/no-number-isnan": 0,
    "es-x/no-number-issafeinteger": 0,
    "es-x/no-number-maxsafeinteger": 0,
    "es-x/no-number-minsafeinteger": 0,
    "es-x/no-number-parsefloat": 0,
    "es-x/no-number-parseint": 0,
    "es-x/no-numeric-separators": 0,
    "es-x/no-object-assign": 0,
    "es-x/no-object-create": 0,
    "es-x/no-object-defineproperties": 0,
    "es-x/no-object-defineproperty": 0,
    "es-x/no-object-entries": 0,
    "es-x/no-object-freeze": 0,
    "es-x/no-object-fromentries": 0,
    "es-x/no-object-getownpropertydescriptor": 0,
    "es-x/no-object-getownpropertydescriptors": 0,
    "es-x/no-object-getownpropertynames": 0,
    "es-x/no-object-getownpropertysymbols": 0,
    "es-x/no-object-getprototypeof": 0,
    "es-x/no-object-groupby": 2,
    "es-x/no-object-hasown": 0,
    "es-x/no-object-is": 0,
    "es-x/no-object-isextensible": 0,
    "es-x/no-object-isfrozen": 0,
    "es-x/no-object-issealed": 0,
    "es-x/no-object-keys": 0,
    "es-x/no-object-map-groupby": 0,
    "es-x/no-object-preventextensions": 0,
    "es-x/no-object-seal": 0,
    "es-x/no-object-setprototypeof": 0,
    "es-x/no-object-super-properties": 0,
    "es-x/no-object-values": 0,
    "es-x/no-octal-numeric-literals": 0,
    "es-x/no-optional-catch-binding": 0,
    "es-x/no-optional-chaining": 0,
    "es-x/no-private-in": 0,
    "es-x/no-promise": 0,
    "es-x/no-promise-all-settled": 0,
    "es-x/no-promise-any": 0,
    "es-x/no-promise-prototype-finally": 0,
    "es-x/no-promise-try": 2,
    "es-x/no-promise-withresolvers": 2,
    "es-x/no-property-shorthands": 0,
    "es-x/no-proxy": 0,
    "es-x/no-reflect": 0,
    "es-x/no-regexp-d-flag": 0,
    "es-x/no-regexp-duplicate-named-capturing-groups": 0,
    "es-x/no-regexp-escape": 2,
    "es-x/no-regexp-lookbehind-assertions": 0,
    "es-x/no-regexp-modifiers": 0,
    "es-x/no-regexp-named-capture-groups": 0,
    "es-x/no-regexp-prototype-compile": 0,
    "es-x/no-regexp-prototype-flags": 0,
    "es-x/no-regexp-s-flag": 0,
    "es-x/no-regexp-u-flag": 0,
    "es-x/no-regexp-unicode-property-escapes": 0,
    "es-x/no-regexp-unicode-property-escapes-2019": 0,
    "es-x/no-regexp-unicode-property-escapes-2020": 0,
    "es-x/no-regexp-unicode-property-escapes-2021": 0,
    "es-x/no-regexp-unicode-property-escapes-2022": 0,
    "es-x/no-regexp-unicode-property-escapes-2023": 0,
    "es-x/no-regexp-v-flag": 0,
    "es-x/no-regexp-y-flag": 0,
    "es-x/no-resizable-and-growable-arraybuffers": 2,
    "es-x/no-rest-parameters": 0,
    "es-x/no-rest-spread-properties": 0,
    "es-x/no-set": 0,
    "es-x/no-set-prototype-difference": 2,
    "es-x/no-set-prototype-intersection": 2,
    "es-x/no-set-prototype-isdisjointfrom": 2,
    "es-x/no-set-prototype-issubsetof": 2,
    "es-x/no-set-prototype-issupersetof": 2,
    "es-x/no-set-prototype-symmetricdifference": 2,
    "es-x/no-set-prototype-union": 2,
    "es-x/no-shadow-catch-param": 0,
    "es-x/no-shared-array-buffer": 2,
    "es-x/no-spread-elements": 0,
    "es-x/no-string-create-html-methods": 0,
    "es-x/no-string-fromcodepoint": 0,
    "es-x/no-string-prototype-at": 0,
    "es-x/no-string-prototype-codepointat": 0,
    "es-x/no-string-prototype-endswith": 0,
    "es-x/no-string-prototype-includes": 0,
    "es-x/no-string-prototype-iswellformed": 2,
    "es-x/no-string-prototype-iswellformed-towellformed": 0,
    "es-x/no-string-prototype-matchall": 0,
    "es-x/no-string-prototype-normalize": 0,
    "es-x/no-string-prototype-padstart-padend": 0,
    "es-x/no-string-prototype-repeat": 0,
    "es-x/no-string-prototype-replaceall": 0,
    "es-x/no-string-prototype-startswith": 0,
    "es-x/no-string-prototype-substr": 0,
    "es-x/no-string-prototype-towellformed": 2,
    "es-x/no-string-prototype-trim": 0,
    "es-x/no-string-prototype-trimleft-trimright": 0,
    "es-x/no-string-prototype-trimstart-trimend": 0,
    "es-x/no-string-raw": 0,
    "es-x/no-subclassing-builtins": 0,
    "es-x/no-suppressederror": 0,
    "es-x/no-symbol": 0,
    "es-x/no-symbol-prototype-description": 0,
    "es-x/no-template-literals": 0,
    "es-x/no-top-level-await": 0,
    "es-x/no-trailing-commas": 0,
    "es-x/no-trailing-dynamic-import-commas": 0,
    "es-x/no-trailing-function-commas": 0,
    "es-x/no-typed-arrays": 0,
    "es-x/no-unicode-codepoint-escapes": 0,
    "es-x/no-using-declarations": 0,
    "es-x/no-weak-map": 0,
    "es-x/no-weak-set": 0,
    "es-x/no-weakrefs": 2,
  },
  "settings": {
    "polyfills": [
      "es:all",
      "Promise",
      "URL",
      "URLSearchParams",
    ],
    "targets": [
      "extends @tinkoff/browserslist-config",
    ],
  },
}
`;
